.circleci/config.yml

```diff
@@ -1,56 +1,378 @@
 # Javascript Node CircleCI 2.0 configuration file
 # Check https://circleci.com/docs/2.0/language-javascript/ for more details
-version: 2
-jobs:
-  build:
-    working_directory: ~/typeorm
+version: 2.1
+
+executors:
+  mariadb:
     docker:
-      # specify pre-built docker images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
       - image: circleci/node:10.15.0
-      - image: mysql:5.7.24
+      - image: mariadb:10.1.37
+        name: mariadb
         environment:
           MYSQL_ROOT_PASSWORD: "admin"
+          MYSQL_USER: "test"
+          MYSQL_PASSWORD: "test"
           MYSQL_DATABASE: "test"
-      - image: mariadb:10.1.37
-        name: mariadb
+  mongodb:
+    docker:
+      - image: circleci/node:10.15.0
+      - image: circleci/mongo:3.4.18
+  mssql:
+    docker:
+      - image: circleci/node:10.15.0
+      - image: "mcr.microsoft.com/mssql/server:2017-GA-ubuntu"
+        environment:
+          SA_PASSWORD: "Admin12345"
+          ACCEPT_EULA: "Y"
+  mysql:
+    docker:
+      - image: circleci/node:10.15.0
+      - image: mysql:5.7.24
         environment:
           MYSQL_ROOT_PASSWORD: "admin"
+          MYSQL_USER: "test"
+          MYSQL_PASSWORD: "test"
           MYSQL_DATABASE: "test"
+  oracle:
+    docker:
+      - image: circleci/node:10.15.0
+      # TODO oracle is skipped right now - not sure what image to use for the DB
+  postgres:
+    docker:
+      - image: circleci/node:10.15.0
       - image: circleci/postgres:9.6.11-postgis
         environment:
           POSTGRES_USER: "test"
           POSTGRES_PASSWORD: "test"
           POSTGRES_DB: "test"
-      - image: circleci/mongo:3.4.18
-      # - image: mcr.microsoft.com/mssql/server:2017-GA-ubuntu
-      #   environment:
-      #     SA_PASSWORD: "Admin12345"
-      #     ACCEPT_EULA: "Y"
-      # - image: cockroachdb/cockroach-unstable:latest
-      #   command: start --insecure
+  sqlite:
+    docker:
+      - image: circleci/node:10.15.0 # sqlite doesn't require another image
+  sqljs:
+    docker:
+      - image: circleci/node:10.15.0 # sqljs doesn't require another image
 
+commands:
```

Comment:
Commands are job `steps` that can have parameters so they can be shared between jobs. In this case, I've created one command for running `npm install` and copying the ormconfig (which I now realize is unnecessary since we copy the DB specific ormconfigs for each test). This step also leverage's CircleCI's `restore_cache` and `save_cache` in order to cache the `npm install` to speed up builds.